pkg/openstack/manila.go

```diff
@@ -120,9 +120,8 @@ func ReconcileManila(ctx context.Context, instance *corev1beta1.OpenStackControl
 			return errors.New("default Manila Share images is unset")
 		}
 
-		if manila.Spec.ManilaShares == nil {
-			manila.Spec.ManilaShares = make(map[string]manilav1.ManilaShareTemplate)
-		}
+		// Discard old list of share services and rebuild it
+		manila.Spec.ManilaShares = make(map[string]manilav1.ManilaShareTemplate)
```

Comment:
I don't think it matters here, but I wanted to ask if this could be problematic in terms of a generated hash and map ordering being nondeterministic.

Reply:
Correct, we do not keep an hash of the deployed shares in the top-level controller, so the order doesn't really matter. The hash is computed only for input (the osp-secret and the generated config) for both the top-level controller and each sub-resource, but the deployed shares do not play a role in the hash.
I agree that in a scenario where we would process the shares and compute the hash with a given criteria, we should sort them (by key?), and make sure we have a consistent mechanism of iterating through them to avoid unexpected results.